cmd/mqtt/main.go

```diff
@@ -122,6 +130,35 @@ func main() {
 		log.Fatalf(err.Error())
 	}
 
+	if cfg.mqttTargetHealthCheck != "" {
+		for i := 1; i <= maxHealthCheckRetries; i++ {
```

Comment:
Since you are already checking the condition below, you can ignore it here `for i:=1; ;i++ {...}`. Also, please consider increasing the sleep period in each iteration by multiplying it by `i`. In that case, `maxHealthCheckRetries` should probably be less than 10 (5?) to prevent long waits. This block can be extracted to a separate function so that we don't pollute the `main`.